src/include/distributed/distributed_planner.h

```diff
@@ -108,6 +129,33 @@ typedef struct RelationRowLock
 } RelationRowLock;
 
 
+typedef struct DistributedPlanningContext
+{
+	/* The parsed query that is given to the planner. It is a slightly modified
+	 * to work with the standard_planner */
+	Query *query;
+
+	/* A copy of the original parsed query that is given to the planner. This
+	 * doesn't contain most of the changes that are made to parse. There's one
+	 * that change that is made for non fast path router queries though, which
+	 * is the assigning of RTE identities using AssignRTEIdentities. This is
+	 * NULL for non distributed plans, since those don't need it. */
+	Query *originalQuery;
+
+	/* the cursor options given to the planner */
+	int cursorOptions;
+
+	/* the ParamListInfo that is given to the planner */
+	ParamListInfo boundParams;
+
+	/* Plan created either by standard_planner or by FastPathPlanner */
```

Comment:
I think this comment is out-dated, we only  store plans generated by standard_planner?

Reply:
No, the plan created by FastPathPlanner is also put in there. `CreateDistributedPlannedStmt` will finalize whatever plan is in there.